File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

from typing import List


# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        position_dict = {}

        for index, num in enumerate(nums):
            remaining = target - num
            
            #Check if needed number exist in position_dict if exist return return answer
            if position_dict.get(remaining, None) is not None:
                return [position_dict.get(remaining), index]
            
            position_dict[num] = index
        return []




# @lc code=end

File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start

import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(int i=0 ; i<nums.length ; i++){
            int complement = target - nums[i];
            if(map.containsKey(complement)){
                return new int[] {map.get(complement), i};
            }
            map.put(nums[i], i);
        }

        return new int[] {0, 0};
        
    }
}
// @lc code=end

